scripts/imgtool/keys/general.py

```diff
@@ -19,17 +19,46 @@ def _public_emit(self, header, trailer, indent, file=sys.stdout, len_format=None
         if len_format is not None:
             print(len_format.format(len(encoded)), file=file)
 
-    def emit_c(self, file=sys.stdout):
+    def _private_emit(self, header, trailer, indent, file=sys.stdout, len_format=None):
+        print(AUTOGEN_MESSAGE, file=file)
+        print(header, end='', file=file)
+        encoded = self.get_private_bytes()
+        for count, b in enumerate(encoded):
+            if count % 8 == 0:
+                print("\n" + indent, end='', file=file)
+            else:
+                print(" ", end='', file=file)
+            print("0x{:02x},".format(b), end='', file=file)
+        print("\n" + trailer, file=file)
+        if len_format is not None:
+            print(len_format.format(len(encoded)), file=file)
+
+    def emit_c_public(self, file=sys.stdout):
         self._public_emit(
                 header="const unsigned char {}_pub_key[] = {{".format(self.shortname()),
                 trailer="};",
                 indent="    ",
                 len_format="const unsigned int {}_pub_key_len = {{}};".format(self.shortname()),
                 file=file)
 
-    def emit_rust(self, file=sys.stdout):
+    def emit_rust_public(self, file=sys.stdout):
         self._public_emit(
                 header="static {}_PUB_KEY: &'static [u8] = &[".format(self.shortname().upper()),
                 trailer="];",
                 indent="    ",
                 file=file)
+
+    def emit_c_private(self, file=sys.stdout):
+        self._private_emit(
+                header="const unsigned char enc_priv_key[] = {",
+                trailer="};",
+                indent="    ",
+                len_format="const unsigned int enc_priv_key_len = {};",
+                file=file)
+
+    def emit_rust_private(self, file=sys.stdout):
```

Comment:
This won't be used by the simulator, so it should be removed.

Reply:
Done